src/BalconyBotanica/Core/Algorithm/Recommender.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using BalconyBotanica.Core.DomainObjects;
+using BalconyBotanica.Hosts.Models;
+
+namespace BalconyBotanica.Core.Algorithm
+
+{
+    /// <summary>
+    /// provides a list of (generic) items when given a set of quiz-answers
+    /// </summary>
+    public class Recommender
+    {
+        // TODO: make Recommend more generic, for now I want it working
+        public RecommendedPlants Recommend(RecommendedPlants insertedArray, QuizAnswers quizAnswers)
+        {
+
+            PlantData[] arrayToFilter = insertedArray.Plants;
+
+            // DONE: check sunlight and delete all the plants who can’t function with that climate
+            // TODO: If answer has "full sun" delete "full shade".
+            // TODO: if awnser had "full shade" delete "full sun".
+            if (quizAnswers.sunlight == Sunlight.FULL_SUN)
```

Comment:
This is a good start and I can see you plan to make improvements on it and it's nice to see what plans you have for it from reading the TODO statements.

Something to consider to improve this code:

What if in the future another Sunlight option was needed, such as `Sunlight.FIFTY_PERCENT_SHADE` and `Sunlight.SEVENTY_FIVE_PERCENT_SHADE`, would you need to keep updating the algorithm with every new `Sunlight` type?
What if in the future another Watering type is needed like `WateringSchedule.DAILY`? Would you need to update the algorithm every time a new Watering type is added?
What if the developer doesn't know they need to update the algorithm? Probably the tests would still pass if a new enum was added but the algorithm might not be working anymore.

There is a way to make this code so that when a new `Sunlight` or whatever type is added, the algorithm code can be left as is and it will work. 

Hint, you need to change the enum to:

```
    public enum Sunlight
    {
        FULL_SUN = 0,
        PART_SHADE = 1,
        FILTERED_SHADE = 2,
        FULL_SHADE = 3
    }
```